src/components/BlogForm/TitleEntry.tsx

```diff
@@ -0,0 +1,112 @@
+"use client";
+import {
+  Button,
+  Container,
+  Stack,
+  TextField,
+  Typography,
+  useTheme,
+} from "@mui/material";
+import { postBlog } from "api/blog";
+import { LoadingScreen } from "components/global-components";
+import _ from "lodash";
+import { useSession } from "next-auth/react";
+import { useSnackbar } from "notistack";
+import { Controller, useForm } from "react-hook-form";
+import { v4 as uuid } from "uuid";
+import { titleDataProps } from "./types";
+
+const TitleEntry = () => {
+  const theme = useTheme();
+  const { enqueueSnackbar } = useSnackbar();
+  const { data: session } = useSession();
+
+  const { control, handleSubmit } = useForm({
+    defaultValues: {
+      title: "",
+      date: "",
+    },
+  });
+
+  if (!session?.user) {
+    return <LoadingScreen />;
+  }
+
+  const { user } = session;
+
+  const onSubmit = (data: titleDataProps) => {
+    const blogSlug = _.kebabCase(data.title);
+    const blogPostId = blogSlug + uuid();
+    postBlog(blogPostId, {
+      ...data,
+      author: user.name,
+      slug: blogSlug,
+      date: new Date().toDateString(),
+    })
+      .then(() => {
+        enqueueSnackbar("Success! A new blog post has been initialised");
+        window.location.replace("http://localhost:3000/dashboard");
```

Comment:
use `router.push()` - we want to stay agnostic of the url domain so when its pushed to `dev.gradguru.app` or `www.gradguru.app` the code doesn't break

Reply:
`router.push('/dashboard')`

Reply:
router.push() doesnt come up as code, tbh ive just removed it as it not needed and i wanted just to push these changes so i can work from here on react quill instead of this branch getting lost.

Reply:
ah you have to import `useRouter()` from next/navigation

`const router = useRouter()`